classes/components/forms/context/PKPDoiRegistrationSettingsForm.php

```diff
@@ -62,7 +63,7 @@ public function __construct(string $action, array $locales, Context $context)
 
         $this->agencyFields = [];
 
-        $registrationAgencies->each(function (IDoiRegistrationAgency $agency) use (&$options, $context) {
+        $registrationAgencies->each(function (IDoiRegistrationAgency|Plugin $agency) use (&$options, $context) {
```

Comment:
@jonasraoni, are you sure this is necessary? It looks to me like this can only be `CrossRefPlugin` or `DatacitePlugin`, both of which are `Plugin`s but also implement `IDoiRegistrationAgency`.

Reply:
The reasoning for this is that some places are also using operations from the `Plugin` class (e.g. `$agency->getName()`).

The most meaningful way would be to use the intersection type `IDoiRegistrationAgency&Plugin`, but it requires PHP 8.1.

Reply:
OK, that makes sense to me.